src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1580,17 +1629,21 @@ namespace winrt::TerminalApp::implementation
             }
         }
 
-        _CloseAllTabs();
+        // Since _RemoveTab is asynchronous, create a snapshot of the  tabs we want to remove
+        std::vector<winrt::TerminalApp::TabBase> tabsToRemove;
+        std::copy(begin(_tabs), end(_tabs), std::back_inserter(tabsToRemove));
+        _RemoveTabs(tabsToRemove);
     }
 
     // Method Description:
-    // - Remove all the tabs opened and the terminal will terminate
-    //   on its own when the last tab is closed.
-    void TerminalPage::_CloseAllTabs()
+    // - Closes provided tabs one by one
+    // Arguments:
+    // - tabs - tabs to remove
+    winrt::fire_and_forget TerminalPage::_RemoveTabs(const std::vector<winrt::TerminalApp::TabBase> tabs)
     {
-        while (_tabs.Size() != 0)
+        for (auto& tab : tabs)
```

Comment:
Just as a sanity check, if we have 4 tabs, and the second tab is read-only, we'll
1. close tab 1 immediately
2. prompt for confirmation to close tab 2. (we may or may not end up closing it)
3. close tab 3 (after we figure out what we're doing with tab 2)
4. close tab 4 immediately after closing 3

Right?

Reply:
The tabs are closed one by one because we await each one of them